optuna/study.py

```diff
@@ -175,6 +175,19 @@ def __setstate__(self, state: Dict[Any, Any]) -> None:
     def user_attrs(self) -> Dict[str, Any]:
         """Return user attributes.
 
+        Example:
+
+            .. testcode::
+
+                import optuna
+
+                study = optuna.create_study()
+
+                study.set_user_attr("contributors", ["Akiba", "Sano"])
+                study.set_user_attr("dataset", "MNIST")
```

Comment:
I think these values are taken from [this tutorial](https://optuna.readthedocs.io/en/stable/tutorial/attributes.html), It may be difficult for readers to imagine the actual usage. I guess they describe the objective function, but this example doesn't have objective functions. So, for example, how about adding a simple objective function and explaining it with the `user_attrs`?

```python
def objective(trial):
    x = trial.suggest_float("x", 0, 1)
    y = trial.suggest_float("y", 0, 1)
    return x ** 2 + y ** 2

...

study.set_user_attr("objective function", "quadratic fuction")
study.set_user_attr("dimensions", 2)
study.set_user_attr("contributors", ["Akiba", "Sano"])
```

Reply:
Ah, I should review #1744 first. I'll copy this comment and ask the other reviewer about it there.